dot/rpc/subscription/websocket.go

```diff
@@ -333,14 +334,14 @@ func (c *WSConn) initExtrinsicWatch(reqID float64, params interface{}) (Listener
 	c.mu.Unlock()
 
 	err = c.CoreAPI.HandleSubmittedExtrinsic(extBytes)
-	if errors.Is(err, runtime.ErrInvalidTransaction) || errors.Is(err, runtime.ErrUnknownTransaction) {
-		c.safeSend(newSubscriptionResponse(authorExtrinsicUpdatesMethod, extSubmitListener.subID, "invalid"))
-		return nil, err
-	} else if err != nil {
-		c.safeSendError(reqID, nil, err.Error())
-		return nil, err
+	if err != nil {
+		if errors.Is(err, runtimererrors.ErrInvalidTxn) || errors.Is(err, runtimererrors.ErrUnknownTxn) {
+			c.safeSend(newSubscriptionResponse(authorExtrinsicUpdatesMethod, extSubmitListener.subID, "invalid"))
+		} else {
+			c.safeSendError(reqID, nil, err.Error())
+		}
```

Comment:
since you have used `switch` statement to handle errors would be nice to use it here as well:
```go
switch {
  case errors.Is(err, runtimererrors.ErrInvalidTxn), 
      errors.Is(err, runtimererrors.ErrUnknownTxn):
    c.safeSend(newSubscriptionResponse(authorExtrinsicUpdatesMethod, extSubmitListener.subID, "invalid"))
  default:
    c.safeSendError(reqID, nil, err.Error())
}
```